Documentation/install/upgrade.rst

```diff
@@ -168,7 +168,9 @@ Kubernetes resources are updated accordingly to version you are upgrading to:
 
   .. group-tab:: Helm
 
-    Deploy Cilium release via Helm:
+    Deploy Cilium release via Helm (Running this command will overwrite the
+    existing cluster's `ConfigMap` which might not be ideal if you want to keep
+    your existing `ConfigMap`.)
```

Comment:
Do we need an overall warning that this will possibly break compatibility as we will use the new helm defaults for upgrades? So far we have been assuming that the ConfigMap values would remain unchanged. The above `helm template` will overwrite it as far as I understand. Our backwards compatibility guarantee has been around not changing the defaults in the binary but default breaking changes to the new value in the ConfigMap so new deployments get the good new values without breaking existing clusters?

Reply:
>Do we need an overall warning that this will possibly break compatibility as we will use the new helm defaults for upgrades?

yes. I think we need to take an overall look at this section since we use `helm upgrade` in a couple places.

>So far we have been assuming that the ConfigMap values would remain unchanged. The above helm template will overwrite it as far as I understand

good point, I've added the warning in the notes section.

>Our backwards compatibility guarantee has been around not changing the defaults in the binary but default breaking changes to the new value in the ConfigMap so new deployments get the good new values without breaking existing clusters?

If I understand you correctly, yes that is supposed to be our guarantee.